core/merge.js

```diff
@@ -548,7 +560,15 @@ class Merge {
 
       const existingDstRev = existingDstRevs[dst._id]
       if (existingDstRev && folder.overwrite) {
-        dst._rev = existingDstRev
+        if (src.deleted) {
+          // Don't create destination that would overwrite existing file
+          continue
+          // TODO: document with test deleting child file then move its parent
+          // folder, overwriting another folder with child file of same name.
+          // The file in the desitnation folder should not be deleted.
```

Comment:
*destination

Reply:
Thanks. And I should implement the test.